### ISSUE TITLE
Support for compose 1.7.+

### ISSUE THREAD

**Issue by prudak-listonic** [microsoft/clarity#716]
Hello, my team and I are working on a project, which uses Compose 1.7.3 (BOM 2024.09.03). It seems this version isn't supported by Clarity yet, causing app screens to appear blank in recordings and heatmaps.
Perhaps there are already plans on adding support for version 1.7.+? If so, could you provide an estimated timeline? And if not, I'd like to request adding it to your roadmap.

**Comment by prudak-listonic** [comment]
@ibradwan @SaadeldinBadr
Hello, I would appreciate any update regarding this issue - is there any chance it will be resolved soon?

**Comment by ibradwan** [comment]
@prudak-listonic yes, we already have a version in the pipeline that supports 1.7-1.8. We'll update this thread once released!

**Comment by prudak-listonic** [comment]
Hello @SaadeldinBadr
Thank you for your answer! I've just found some time to test the newest version, but unfortunately I've encountered another issue.

Clarity-compose gets downloaded during the gradle sync without any problems, but only up to version 3.0.1. It seems I can't download any of the two newest versions (3.0.2 and 3.1.0).
The gradle sync in this case behaves as if those versions didn't exist  - it finishes without any errors, yet I can't find clarity among external modules (I've tried to download a non-existing 3.2.0 version with the same result).

Do you have any ideas what might be the cause?

**Comment by ibradwan** [comment]
Hi @prudak-listonic,

Please try to add this to your app build.gradle file under `android > defaultConfig` section:
```
android {
    defaultConfig {
        ....
        missingDimensionStrategy("environment", "prod")
    }
}
```

This is a stopgap until we fix this issue.

**Comment by SaadeldinBadr** [comment]
Hello @prudak-listonic,
We have just released “clarity-compose :3.1.1” that should fix the `failed to resolve` issue, please give it a try and let us know if it works for you.

**Comment by prudak-listonic** [comment]
Hi @SaadeldinBadr,
Yes, now it works like a charm, thank you very much!
